### ISSUE TITLE
Excessive logging from gpu-operator

### ISSUE THREAD

**Issue by silashansen** [NVIDIA/gpu-operator#122]
### 1. Quick Debug Checklist
- [x] Are you running on an Ubuntu 18.04 node?
- [x] Are you running Kubernetes v1.13+?
- [x] Are you running Docker (>= 18.06) or CRIO (>= 1.13+)?

### 1. Issue or feature description

The operator is working, but the gpu-operator pod is currently writing about 1000-2000 message like these, per second.
It's all INFO level, so it doesn't seem to stem from an error.

It's causing things to break.

Can I turn off logging or change the log-level somehow?

```
{"level":"info","ts":1609192433.581212,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","Phase":"Succeeded","==":"Succeeded"}
{"level":"info","ts":1609192433.5845795,"logger":"controller_clusterpolicy","msg":"Found Resource","ServiceAccount":"nvidia-dcgm-exporter","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.5916452,"logger":"controller_clusterpolicy","msg":"Found Resource","Role":"nvidia-dcgm-exporter","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.5951428,"logger":"controller_clusterpolicy","msg":"Found Resource","RoleBinding":"nvidia-dcgm-exporter","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.6094966,"logger":"controller_clusterpolicy","msg":"Found Resource","Service":"nvidia-dcgm-exporter","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.6096227,"logger":"controller_clusterpolicy","msg":"5.4.0-1029-aws","Request.Namespace":"default","Request.Name":"Node"}
{"level":"info","ts":1609192433.6296344,"logger":"controller_clusterpolicy","msg":"Found Resource","DaemonSet":"nvidia-dcgm-exporter","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.6296606,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","LabelSelector":"app=nvidia-dcgm-exporter"}
{"level":"info","ts":1609192433.6297297,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberOfDaemonSets":1}
{"level":"info","ts":1609192433.629738,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberUnavailable":0}
{"level":"info","ts":1609192433.6297438,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","LabelSelector":"app=nvidia-dcgm-exporter"}
{"level":"info","ts":1609192433.629824,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","NumberOfPods":1}
{"level":"info","ts":1609192433.6298301,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","Phase":"Running","==":"Running"}
{"level":"info","ts":1609192433.6349864,"logger":"controller_clusterpolicy","msg":"Found Resource","ServiceAccount":"nvidia-gpu-feature-discovery","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.6819491,"logger":"controller_clusterpolicy","msg":"Found Resource","DaemonSet":"gpu-feature-discovery","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.6819775,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","LabelSelector":"app=gpu-feature-discovery"}
{"level":"info","ts":1609192433.6820583,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberOfDaemonSets":1}
{"level":"info","ts":1609192433.682067,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberUnavailable":0}
{"level":"info","ts":1609192433.682073,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","LabelSelector":"app=gpu-feature-discovery"}
{"level":"info","ts":1609192433.6821322,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","NumberOfPods":0}
{"level":"info","ts":1609192433.6882071,"logger":"controller_clusterpolicy","msg":"ClusterPolicy step wasn't ready","State:":"notReady"}
{"level":"info","ts":1609192433.6882668,"logger":"controller_clusterpolicy","msg":"Reconciling ClusterPolicy","Request.Name":"cluster-policy"}
{"level":"info","ts":1609192433.689996,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/state-driver"}
{"level":"info","ts":1609192433.6901777,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"ServiceAccount","in path:":"/opt/gpu-operator/state-driver"}
{"level":"info","ts":1609192433.6902907,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Role","in path:":"/opt/gpu-operator/state-driver"}
{"level":"info","ts":1609192433.6904204,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"RoleBinding","in path:":"/opt/gpu-operator/state-driver"}
{"level":"info","ts":1609192433.6905317,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"ConfigMap","in path:":"/opt/gpu-operator/state-driver"}
{"level":"info","ts":1609192433.6906319,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"DaemonSet","in path:":"/opt/gpu-operator/state-driver"}
{"level":"info","ts":1609192433.6910417,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/state-container-toolkit"}
{"level":"info","ts":1609192433.6911848,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"ServiceAccount","in path:":"/opt/gpu-operator/state-container-toolkit"}
{"level":"info","ts":1609192433.691283,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Role","in path:":"/opt/gpu-operator/state-container-toolkit"}
{"level":"info","ts":1609192433.6913946,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"RoleBinding","in path:":"/opt/gpu-operator/state-container-toolkit"}
{"level":"info","ts":1609192433.6915073,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"DaemonSet","in path:":"/opt/gpu-operator/state-container-toolkit"}
{"level":"info","ts":1609192433.6918828,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/state-driver-validation"}
{"level":"info","ts":1609192433.6919606,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Pod","in path:":"/opt/gpu-operator/state-driver-validation"}
{"level":"info","ts":1609192433.692125,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/state-device-plugin"}
{"level":"info","ts":1609192433.692244,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"ServiceAccount","in path:":"/opt/gpu-operator/state-device-plugin"}
{"level":"info","ts":1609192433.692383,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Role","in path:":"/opt/gpu-operator/state-device-plugin"}
{"level":"info","ts":1609192433.6925635,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"RoleBinding","in path:":"/opt/gpu-operator/state-device-plugin"}
{"level":"info","ts":1609192433.6927488,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"DaemonSet","in path:":"/opt/gpu-operator/state-device-plugin"}
{"level":"info","ts":1609192433.6931162,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/state-device-plugin-validation"}
{"level":"info","ts":1609192433.6932118,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Pod","in path:":"/opt/gpu-operator/state-device-plugin-validation"}
{"level":"info","ts":1609192433.6943107,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/state-monitoring"}
{"level":"info","ts":1609192433.6944973,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"ServiceAccount","in path:":"/opt/gpu-operator/state-monitoring"}
{"level":"info","ts":1609192433.6945999,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Role","in path:":"/opt/gpu-operator/state-monitoring"}
{"level":"info","ts":1609192433.6947455,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"RoleBinding","in path:":"/opt/gpu-operator/state-monitoring"}
{"level":"info","ts":1609192433.6948936,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Service","in path:":"/opt/gpu-operator/state-monitoring"}
{"level":"info","ts":1609192433.695047,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"DaemonSet","in path:":"/opt/gpu-operator/state-monitoring"}
{"level":"info","ts":1609192433.6954627,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/gpu-feature-discovery"}
{"level":"info","ts":1609192433.695571,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"ServiceAccount","in path:":"/opt/gpu-operator/gpu-feature-discovery"}
{"level":"info","ts":1609192433.6956348,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"DaemonSet","in path:":"/opt/gpu-operator/gpu-feature-discovery"}
{"level":"info","ts":1609192433.7034113,"logger":"controller_clusterpolicy","msg":"Found Resource","ServiceAccount":"nvidia-driver","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.7100425,"logger":"controller_clusterpolicy","msg":"Found Resource","Role":"nvidia-driver","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.714254,"logger":"controller_clusterpolicy","msg":"Found Resource","RoleBinding":"nvidia-driver","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.7220492,"logger":"controller_clusterpolicy","msg":"Found Resource","ConfigMap":"nvidia-driver","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.7221723,"logger":"controller_clusterpolicy","msg":"5.4.0-1029-aws","Request.Namespace":"default","Request.Name":"Node"}
{"level":"info","ts":1609192433.7299008,"logger":"controller_clusterpolicy","msg":"Found Resource","DaemonSet":"nvidia-driver-daemonset","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.7299287,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","LabelSelector":"app=nvidia-driver-daemonset"}
{"level":"info","ts":1609192433.7300084,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberOfDaemonSets":1}
{"level":"info","ts":1609192433.7300155,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberUnavailable":0}
{"level":"info","ts":1609192433.7300215,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","LabelSelector":"app=nvidia-driver-daemonset"}
{"level":"info","ts":1609192433.730102,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","NumberOfPods":1}
{"level":"info","ts":1609192433.730109,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","Phase":"Running","==":"Running"}
{"level":"info","ts":1609192433.739908,"logger":"controller_clusterpolicy","msg":"Found Resource","ServiceAccount":"nvidia-container-toolkit","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.742572,"logger":"controller_clusterpolicy","msg":"Found Resource","Role":"nvidia-container-toolkit","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.7488923,"logger":"controller_clusterpolicy","msg":"Found Resource","RoleBinding":"nvidia-container-toolkit","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.7489746,"logger":"controller_clusterpolicy","msg":"Info: Could not find environment variable RUNTIME in container nvidia-container-toolkit-ctr, appending it"}
{"level":"info","ts":1609192433.7838814,"logger":"controller_clusterpolicy","msg":"Found Resource","DaemonSet":"nvidia-container-toolkit-daemonset","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.7839112,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","LabelSelector":"app=nvidia-container-toolkit-daemonset"}
{"level":"info","ts":1609192433.7839894,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberOfDaemonSets":1}
{"level":"info","ts":1609192433.784,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberUnavailable":0}
{"level":"info","ts":1609192433.7840056,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","LabelSelector":"app=nvidia-container-toolkit-daemonset"}
{"level":"info","ts":1609192433.7840955,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","NumberOfPods":1}
{"level":"info","ts":1609192433.7841022,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","Phase":"Running","==":"Running"}
{"level":"info","ts":1609192433.7901073,"logger":"controller_clusterpolicy","msg":"Found Resource","Pod":"nvidia-driver-validation","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.7901354,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","LabelSelector":"app=nvidia-driver-validation"}
{"level":"info","ts":1609192433.7902684,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","NumberOfPods":1}
{"level":"info","ts":1609192433.7902758,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","Phase":"Succeeded","==":"Succeeded"}
{"level":"info","ts":1609192433.7952373,"logger":"controller_clusterpolicy","msg":"Found Resource","ServiceAccount":"nvidia-device-plugin","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.8001258,"logger":"controller_clusterpolicy","msg":"Found Resource","Role":"nvidia-device-plugin","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.8059485,"logger":"controller_clusterpolicy","msg":"Found Resource","RoleBinding":"nvidia-device-plugin","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.8273447,"logger":"controller_clusterpolicy","msg":"Found Resource","DaemonSet":"nvidia-device-plugin-daemonset","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.827372,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","LabelSelector":"app=nvidia-device-plugin-daemonset"}
{"level":"info","ts":1609192433.82744,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberOfDaemonSets":1}
{"level":"info","ts":1609192433.8274503,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberUnavailable":0}
{"level":"info","ts":1609192433.8274567,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","LabelSelector":"app=nvidia-device-plugin-daemonset"}
{"level":"info","ts":1609192433.82753,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","NumberOfPods":1}
{"level":"info","ts":1609192433.82754,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","Phase":"Running","==":"Running"}
{"level":"info","ts":1609192433.831145,"logger":"controller_clusterpolicy","msg":"Found Resource","Pod":"nvidia-device-plugin-validation","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.8311682,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","LabelSelector":"app=nvidia-device-plugin-validation"}
{"level":"info","ts":1609192433.8312619,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","NumberOfPods":1}
{"level":"info","ts":1609192433.8312747,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","Phase":"Succeeded","==":"Succeeded"}
{"level":"info","ts":1609192433.8384423,"logger":"controller_clusterpolicy","msg":"Found Resource","ServiceAccount":"nvidia-dcgm-exporter","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.8416128,"logger":"controller_clusterpolicy","msg":"Found Resource","Role":"nvidia-dcgm-exporter","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.8496084,"logger":"controller_clusterpolicy","msg":"Found Resource","RoleBinding":"nvidia-dcgm-exporter","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.8612053,"logger":"controller_clusterpolicy","msg":"Found Resource","Service":"nvidia-dcgm-exporter","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.8613372,"logger":"controller_clusterpolicy","msg":"5.4.0-1029-aws","Request.Namespace":"default","Request.Name":"Node"}
{"level":"info","ts":1609192433.878421,"logger":"controller_clusterpolicy","msg":"Found Resource","DaemonSet":"nvidia-dcgm-exporter","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.8784468,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","LabelSelector":"app=nvidia-dcgm-exporter"}
{"level":"info","ts":1609192433.8785145,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberOfDaemonSets":1}
{"level":"info","ts":1609192433.8785217,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberUnavailable":0}
{"level":"info","ts":1609192433.8785272,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","LabelSelector":"app=nvidia-dcgm-exporter"}
{"level":"info","ts":1609192433.878627,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","NumberOfPods":1}
{"level":"info","ts":1609192433.878633,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","Phase":"Running","==":"Running"}
{"level":"info","ts":1609192433.8879645,"logger":"controller_clusterpolicy","msg":"Found Resource","ServiceAccount":"nvidia-gpu-feature-discovery","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.9291635,"logger":"controller_clusterpolicy","msg":"Found Resource","DaemonSet":"gpu-feature-discovery","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.9291933,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","LabelSelector":"app=gpu-feature-discovery"}
{"level":"info","ts":1609192433.9292598,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberOfDaemonSets":1}
{"level":"info","ts":1609192433.9292703,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberUnavailable":0}
{"level":"info","ts":1609192433.929277,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","LabelSelector":"app=gpu-feature-discovery"}
{"level":"info","ts":1609192433.929334,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","NumberOfPods":0}
{"level":"info","ts":1609192433.936357,"logger":"controller_clusterpolicy","msg":"ClusterPolicy step wasn't ready","State:":"notReady"}
{"level":"info","ts":1609192433.9364161,"logger":"controller_clusterpolicy","msg":"Reconciling ClusterPolicy","Request.Name":"cluster-policy"}
{"level":"info","ts":1609192433.9404318,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/state-driver"}
{"level":"info","ts":1609192433.9406538,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"ServiceAccount","in path:":"/opt/gpu-operator/state-driver"}
{"level":"info","ts":1609192433.9407766,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Role","in path:":"/opt/gpu-operator/state-driver"}
{"level":"info","ts":1609192433.940911,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"RoleBinding","in path:":"/opt/gpu-operator/state-driver"}
{"level":"info","ts":1609192433.9410658,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"ConfigMap","in path:":"/opt/gpu-operator/state-driver"}
{"level":"info","ts":1609192433.9411936,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"DaemonSet","in path:":"/opt/gpu-operator/state-driver"}
{"level":"info","ts":1609192433.9416785,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/state-container-toolkit"}
{"level":"info","ts":1609192433.9418375,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"ServiceAccount","in path:":"/opt/gpu-operator/state-container-toolkit"}
{"level":"info","ts":1609192433.9419203,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Role","in path:":"/opt/gpu-operator/state-container-toolkit"}
{"level":"info","ts":1609192433.9420428,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"RoleBinding","in path:":"/opt/gpu-operator/state-container-toolkit"}
{"level":"info","ts":1609192433.9421885,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"DaemonSet","in path:":"/opt/gpu-operator/state-container-toolkit"}
{"level":"info","ts":1609192433.9427235,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/state-driver-validation"}
{"level":"info","ts":1609192433.9428217,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Pod","in path:":"/opt/gpu-operator/state-driver-validation"}
{"level":"info","ts":1609192433.9430265,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/state-device-plugin"}
{"level":"info","ts":1609192433.9431689,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"ServiceAccount","in path:":"/opt/gpu-operator/state-device-plugin"}
{"level":"info","ts":1609192433.9432428,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Role","in path:":"/opt/gpu-operator/state-device-plugin"}
{"level":"info","ts":1609192433.9433525,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"RoleBinding","in path:":"/opt/gpu-operator/state-device-plugin"}
{"level":"info","ts":1609192433.9435024,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"DaemonSet","in path:":"/opt/gpu-operator/state-device-plugin"}
{"level":"info","ts":1609192433.9438758,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/state-device-plugin-validation"}
{"level":"info","ts":1609192433.9439728,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Pod","in path:":"/opt/gpu-operator/state-device-plugin-validation"}
{"level":"info","ts":1609192433.944199,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/state-monitoring"}
{"level":"info","ts":1609192433.9444606,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"ServiceAccount","in path:":"/opt/gpu-operator/state-monitoring"}
{"level":"info","ts":1609192433.9445925,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Role","in path:":"/opt/gpu-operator/state-monitoring"}
{"level":"info","ts":1609192433.9447803,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"RoleBinding","in path:":"/opt/gpu-operator/state-monitoring"}
{"level":"info","ts":1609192433.9449728,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"Service","in path:":"/opt/gpu-operator/state-monitoring"}
{"level":"info","ts":1609192433.9451652,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"DaemonSet","in path:":"/opt/gpu-operator/state-monitoring"}
{"level":"info","ts":1609192433.9456894,"logger":"controller_clusterpolicy","msg":"Getting assets from: ","path:":"/opt/gpu-operator/gpu-feature-discovery"}
{"level":"info","ts":1609192433.9458497,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"ServiceAccount","in path:":"/opt/gpu-operator/gpu-feature-discovery"}
{"level":"info","ts":1609192433.9459438,"logger":"controller_clusterpolicy","msg":"DEBUG: Looking for ","Kind":"DaemonSet","in path:":"/opt/gpu-operator/gpu-feature-discovery"}
{"level":"info","ts":1609192433.9520078,"logger":"controller_clusterpolicy","msg":"Found Resource","ServiceAccount":"nvidia-driver","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.9568298,"logger":"controller_clusterpolicy","msg":"Found Resource","Role":"nvidia-driver","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.963126,"logger":"controller_clusterpolicy","msg":"Found Resource","RoleBinding":"nvidia-driver","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.9662175,"logger":"controller_clusterpolicy","msg":"Found Resource","ConfigMap":"nvidia-driver","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.9663436,"logger":"controller_clusterpolicy","msg":"5.4.0-1029-aws","Request.Namespace":"default","Request.Name":"Node"}
{"level":"info","ts":1609192433.9836533,"logger":"controller_clusterpolicy","msg":"Found Resource","DaemonSet":"nvidia-driver-daemonset","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.9836864,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","LabelSelector":"app=nvidia-driver-daemonset"}
{"level":"info","ts":1609192433.9837842,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberOfDaemonSets":1}
{"level":"info","ts":1609192433.983792,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberUnavailable":0}
{"level":"info","ts":1609192433.9837992,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","LabelSelector":"app=nvidia-driver-daemonset"}
{"level":"info","ts":1609192433.9838743,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","NumberOfPods":1}
{"level":"info","ts":1609192433.9838808,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","Phase":"Running","==":"Running"}
{"level":"info","ts":1609192433.9944205,"logger":"controller_clusterpolicy","msg":"Found Resource","ServiceAccount":"nvidia-container-toolkit","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.9971688,"logger":"controller_clusterpolicy","msg":"Found Resource","Role":"nvidia-container-toolkit","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.9998164,"logger":"controller_clusterpolicy","msg":"Found Resource","RoleBinding":"nvidia-container-toolkit","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192433.9998722,"logger":"controller_clusterpolicy","msg":"Info: Could not find environment variable RUNTIME in container nvidia-container-toolkit-ctr, appending it"}
{"level":"info","ts":1609192434.0319412,"logger":"controller_clusterpolicy","msg":"Found Resource","DaemonSet":"nvidia-container-toolkit-daemonset","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192434.0319717,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","LabelSelector":"app=nvidia-container-toolkit-daemonset"}
{"level":"info","ts":1609192434.0320454,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberOfDaemonSets":1}
{"level":"info","ts":1609192434.032054,"logger":"controller_clusterpolicy","msg":"DEBUG: DaemonSet","NumberUnavailable":0}
{"level":"info","ts":1609192434.032061,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","LabelSelector":"app=nvidia-container-toolkit-daemonset"}
{"level":"info","ts":1609192434.0321448,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","NumberOfPods":1}
{"level":"info","ts":1609192434.0321574,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","Phase":"Running","==":"Running"}
{"level":"info","ts":1609192434.037819,"logger":"controller_clusterpolicy","msg":"Found Resource","Pod":"nvidia-driver-validation","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192434.0378618,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","LabelSelector":"app=nvidia-driver-validation"}
{"level":"info","ts":1609192434.0379791,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","NumberOfPods":1}
{"level":"info","ts":1609192434.0379875,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","Phase":"Succeeded","==":"Succeeded"}
{"level":"info","ts":1609192434.0436885,"logger":"controller_clusterpolicy","msg":"Found Resource","ServiceAccount":"nvidia-device-plugin","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192434.0475013,"logger":"controller_clusterpolicy","msg":"Found Resource","Role":"nvidia-device-plugin","Namespace":"gpu-operator-resources"}
{"level":"info","ts":1609192434.0511434,"logger":"controller_clusterpolicy","msg":"Found Resource","RoleBinding":"nvidia-device-plugin","Namespace":"gpu-operator-resources"}
```

**Comment by shivamerla** [comment]
@silashansen This can happen if any of the resources deployed by GPU operator are not ready. Do you see gpu-feature-discovery pods running in `gpu-operator-resources` namespace? Can you describe that daemonset to see why pods cannot be started?

```
{"level":"info","ts":1609192433.929277,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","LabelSelector":"app=gpu-feature-discovery"}
{"level":"info","ts":1609192433.929334,"logger":"controller_clusterpolicy","msg":"DEBUG: Pod","NumberOfPods":0}
{"level":"info","ts":1609192433.936357,"logger":"controller_clusterpolicy","msg":"ClusterPolicy step wasn't ready","State:":"notReady"}
```

**Comment by silashansen** [comment]
@shivamerla Not that I can see. This is the running pods output:

```
> kubectl get pods -A | grep gpu

NAMESPACE                NAME                                                              READY   STATUS      RESTARTS   AGE
default                  gpu-operator-1609238990-node-feature-discovery-master-85c8q22c9   1/1     Running     0          10m
default                  gpu-operator-1609238990-node-feature-discovery-worker-xxpnc       1/1     Running     0          6m31s
default                  gpu-operator-657579678c-8ltgj                                     1/1     Running     6          10m
gpu-operator-resources   nvidia-container-toolkit-daemonset-q4rsn                          1/1     Running     0          3m53s
gpu-operator-resources   nvidia-dcgm-exporter-zb9xv                                        1/1     Running     0          2m39s
gpu-operator-resources   nvidia-device-plugin-daemonset-q6dfh                              1/1     Running     0          2m55s
gpu-operator-resources   nvidia-device-plugin-validation                                   0/1     Completed   0          2m45s
gpu-operator-resources   nvidia-driver-daemonset-wfmgp                                     1/1     Running     0          4m35s
gpu-operator-resources   nvidia-driver-validation                                          0/1     Completed   3          3m44s
```

**Comment by shivamerla** [comment]
@silashansen  Can you describe the `gpu-feature-discovery' daemonset and see why pods are not getting deployed? Are there taints on the node which are not tolerated by just gpu-feature-discovery pods?

**Comment by silashansen** [comment]
Ah, good catch.

I got thrown off because I kept comparing my running pods to the list outlined on: https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/getting-started.html#install-the-gpu-operator, just below where it says "Check the status of the pods to ensure all the containers are running:", which seems to be missing the *gpu-feature-discovery* daemonset pods.

As this is a hybrid cluster, I added the following tolerations to my values.yaml and that seems to have fixed it:
(for others reading this, the specific tolerations will vary depending on your node taints)

```
gfd:
  tolerations:
  - key: nvidia.com/gpu
    operator: Exists
    effect: NoSchedule
```

Thanks!
